File: 20230623/detect_terminal.c
#include <iostream>
#include <unistd.h>

int main() {
    if (isatty(fileno(stdin)) != 0) {
        std::cout << "入力" << std::endl;
    }

    if (isatty(fileno(stdout)) != 0) {
        std::cout << "出力" << std::endl;
    }

    if (isatty(fileno(stderr)) != 0) {
        std::cout << "エラー" << std::endl;
    }
}

// 2XXXX1@ISE88u:~/Practical-Programming/20230623$ g++ detect_terminal.c && ./a.out
// 入力
// 出力
// エラー
// 2XXXX1@ISE88u:~/Practical-Programming/20230623$ g++ detect_terminal.c && ls | ./a.out
// 出力
// エラー

File: 20230623/detect_terminal.py
import sys

if sys.stdin.isatty():
    print('入力')

if sys.stdout.isatty():
    print('出力')

if sys.stderr.isatty():
    print('エラー')

# 2XXXX1@ISE88u:~/Practical-Programming/20230623$ python detect_terminal.py
# 入力
# 出力
# エラー
# 2XXXX1@ISE88u:~/Practical-Programming/20230623$ ls | python detect_terminal.py
# 出力
# エラー
